lib/surface/compiler/parse_tree_translator.ex

```diff
@@ -164,12 +174,12 @@ defmodule Surface.Compiler.ParseTreeTranslator do
     %{tag_name: name}
   end
 
-  def context_for_subblock(_name, _meta, _state, _parent_context) do
-    nil
+  def context_for_subblock(_name, _meta, _state, parent_context) do
+    parent_context
   end
 
-  def context_for_block(_name, _meta, _state) do
-    nil
+  def context_for_block(name, _meta, _state) do
+    %{parent_block: name}
   end
```

Comment:
a slight alternative:

```
def context_for_subblock(name, _meta, _state, parent_context) do
  %{sub_block: name, parent_block: Map.get(parent_context, :block_name)}
end

def context_for_block(name, _meta, _state_ do
  %{block_name: name}
end
```

my reasoning is that I think the attribute `parent_block` makes sense in the subblock context, I'm not sure it does for the block context